src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -1862,6 +1866,48 @@ void GraphOptimizer::FuseMVNAndSimpleOperation(Graph& graph) {
     }
 }
 
+void GraphOptimizer::FuseGatherAndConvert(Graph& graph) {
+    auto& graphNodes = graph.GetNodes();
+
+    auto isSuitableParentNode = [](const NodePtr& node) {
+        return (node->getType() == Type::Gather) && (node->getChildEdges().size() == 1) &&
+               one_of(node->getOriginalInputPrecisionAtPort(0), element::f16, element::bf16);
+    };
+
+    auto parent = graphNodes.begin();
+    while (parent != graphNodes.end()) {
+        auto parentNode = *parent;
+        if (!isSuitableParentNode(parentNode)) {
+            parent++;
+            continue;
+        }
+
+        CPU_GRAPH_OPTIMIZER_SCOPE(FuseGatherAndConvert);
+
+        auto childNode = parentNode->getChildEdgeAt(0)->getChild();
+        if (!parentNode->canFuse(childNode)) {
+            parent++;
+            continue;
+        }
+
+        childNode->fuseInto(parentNode);
+
+        if (childNode->getType() == Type::Convert) {
```

Comment:
We fuse only convert in this pass, don't we?
1. Why don't we check that the child node is convert before the fusion?
2. If we had the check upper, this check would be unnecessary

Reply:
Can we anyway add a separate check on `Convert` in this pass? For now, `canFuse` returns true only for converts, but this transformation can unexpectedly be applied to non-convert nodes if the Gather node supports other fusings in the future

Reply:
sure, have added the check. Thanks Vladislav!